services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaService.kt

```diff
@@ -281,6 +283,31 @@ class HankeKayttajaService(
             } ?: throw HankeNotFoundException(hankeTunnus)
     }
 
+    @Transactional
+    fun updateContactInfo(hankeTunnus: String, update: ContactUpdate, userId: UUID): HankeKayttaja {
+        hankeRepository
+            .findOneByHankeTunnus(hankeTunnus)
+            ?.let { getKayttajaForHanke(userId, it.id) }
+            ?.let { hankeKayttajaEntity ->
+                // changing name is not allowed if the user is identified (has a permission)
+                if (
+                    hankeKayttajaEntity.permission != null &&
+                        (!update.etunimi.isNullOrEmpty() || !update.sukunimi.isNullOrEmpty())
+                ) {
+                    throw UserAlreadyHasPermissionException(
+                        userId.toString(),
+                        hankeKayttajaEntity.id,
+                        hankeKayttajaEntity.permission!!.id
+                    )
+                }
+                hankeKayttajaEntity.sahkoposti = update.sahkoposti
+                hankeKayttajaEntity.puhelin = update.puhelinnumero
+                update.etunimi?.let { hankeKayttajaEntity.etunimi = it }
+                update.sukunimi?.let { hankeKayttajaEntity.sukunimi = it }
```

Comment:
I think nothing is stopping the user from changing their name to an empty string or a single whitespace character.

Reply:
I added check for blanks